manager/utils.py

```diff
@@ -540,3 +548,86 @@ def flush_transaction():
     this function at the appropriate moment
     """
     transaction.commit()
+
+
+def get_report(name, **kwargs):
+    """
+    Returns report data
+
+    :param name: The name of the report to run
+    :param **kwargs: The reports keyword arguments
+    :returns: A tuple of aggregated stat data and a QuerySet of data
+    """
+    if name == 'url_report':
+        return url_report(**kwargs)
+
+    return (None, None)
+
+def url_report(**kwargs):
+    """
+    Returns aggregated instance data and a QuerySet of URLs
+
+    :param **kwargs: A Dictionary of arguments for the report
+    """
+    urls = URL.objects.all()
+    instances = Instance.objects.all()
+
+    email_select = kwargs['email_select'] if 'email_select' in kwargs else None
+    start_date = kwargs['start_date'] if 'start_date' in kwargs else None
+    end_date = kwargs['end_date'] if 'end_date' in kwargs else None
+    day_of_week = kwargs['day_of_week'] if 'day_of_week' in kwargs else None
+    url_filter = kwargs['url_filter'] if 'url_filter' in kwargs else None
+    email_domain = kwargs['email_domain'] if 'email_domain' in kwargs else None
+
+    # Short curcuit the whole thing if required fields aren't set
+    if email_select is None:
+        return (None, None)
+
+    # Get instance stats
+    instances_stats = {}
+    if email_select is not None:
+        instances = instances.filter(email__in=email_select)
+
+    if start_date and end_date is not None:
```

Comment:
Probably want to explicitly check for a `None` value on both `start_date` and `end_date` here, e.g.
`if start_date is not None and end_date is not None:`